### PR TITLE
issue: 1582417 Add Packet Pacing support for upstream

### PR DESCRIPTION
**Pull request by liranoz12**

### CODE REVIEW

**Comment by liranoz12** [comment]
bot:retest

**Comment by liranoz12** [comment]
bot:retest

File: src/vma/ib/base/verbs_extra.h [hunk 1]
```diff
@@ -241,6 +241,20 @@ typedef struct ibv_modify_cq_attr               vma_ibv_cq_attr;
 #define VMA_IBV_CQ_MODERATION                   IBV_CQ_ATTR_MODERATE
 #endif
 
+#ifdef DEFINED_IBV_PACKET_PACING_CAPS
+#define VMA_IBV_QP_RATE_LIMIT                IBV_QP_RATE_LIMIT
+#define vma_is_pacing_caps_supported(attr)   (attr->packet_pacing_caps.qp_rate_limit_min)
```
Comment (rafiw) [inline src/vma/ib/base/verbs_extra.h]: indentation

File: src/vma/ib/base/verbs_extra.h [hunk 2]
```diff
@@ -241,6 +241,20 @@ typedef struct ibv_modify_cq_attr               vma_ibv_cq_attr;
...
+typedef struct ibv_packet_pacing_caps	     vma_ibv_packet_pacing_caps;
+typedef struct ibv_qp_rate_limit_attr                   vma_ibv_rate_limit_attr;
+#define vma_ibv_modify_qp_rate_limit(qp, attr, mask)    ibv_modify_qp_rate_limit(qp, attr)
+#define vma_ibv_init_qps_attr(qp_attr)                  { NOT_IN_USE(qp_attr); }
+#endif // DEFINED_IBV_PACKET_PACING_CAPS
+
+#ifdef DEFINED_IBV_QP_SUPPORT_BURST
+#define vma_is_burst_supported(attr)                    (attr->packet_pacing_caps.qp_rate_limit_min)
```
Comment (rafiw) [inline src/vma/ib/base/verbs_extra.h]: use cap_flags & MLX5_IB_PP_SUPPORT_BURST

File: src/vma/ib/base/verbs_extra.h [hunk 3]
```diff
@@ -392,6 +406,20 @@ typedef struct ibv_exp_flow_spec_action_tag             vma_ibv_flow_spec_action
 typedef struct ibv_exp_flow_spec_action_tag_dummy {}    vma_ibv_flow_spec_action_tag;
 #endif //DEFINED_IBV_FLOW_TAG
 
+#ifdef DEFINED_IBV_PACKET_PACING_CAPS
+#define VMA_IBV_QP_RATE_LIMIT                IBV_EXP_QP_RATE_LIMIT
```
Comment (rafiw) [inline src/vma/ib/base/verbs_extra.h]: indentation

**Comment by liranoz12** [comment]
bot:retest

**Comment by liranoz12** [comment]
bot:retest

**Pull request merged**